### ISSUE TITLE
Add Tag manager to FUG website

### ISSUE THREAD

**Issue by PatReynolds** [FreeUKGen/Coordination#868]
needs #867

**Comment by DeniseColbert** [comment]
Emailed Rich to advise Bev 10/10

**Comment by PatReynolds** [comment]
Conversation on Slack asking if Scott should help out here.

**Comment by DeniseColbert** [comment]
Added Gray (and removed Bev, at her request)

**Comment by arswright-alt-uk** [comment]
@DeniseColbert Can this functionality wait until switching the CMS? (I know WordPress includes full taxonomy management "out of the box")

**Comment by DeniseColbert** [comment]
How soon might we be in that position, @arswright-alt-uk?

**Comment by arswright-alt-uk** [comment]
@DeniseColbert Cancel that - I thought this referred to CMS taxonomy management (just now noticed the "Analytics" tag) - we can proceed straightaway, just need the Google Tag Manager installation code.

Recommend adding Google Analytics (+ other desired tags as applicable) and publishing the container **before** retrieving the installation code to ensure things behave as expected when code is installed.

**Comment by DeniseColbert** [comment]
I've given you access to GTM: Read for containers. Let me know if you need more/different.

**Comment by DeniseColbert** [comment]
Publish access given, if you're happy to proceed. The ID matches the correct account, I think it's good to go.

**Comment by richpomfret** [comment]
@DeniseColbert @arswright-alt-uk is this now live? In terms of GDPR, no marketing tags should be fired until cookie consent is given (or set to 'true' depending upon how Bev implemented it). IF tags are fired without consent, the quickest fix may be to ensure anonymize Ip setting is switched on in Tag manager (I believe we have it in Analytics but Tag manager might affect that).

**Comment by arswright-alt-uk** [comment]
@richpomfret Not live yet - GDPR was one of several considerations I need to address in testing (also need to mirror custom Google Analytics functionality in Google Tag Manager, confirm cross-domain tracking works, etc). Will update this task w/progress.

**Comment by arswright-alt-uk** [comment]
@richpomfret **Short answer:** Looks OK at present (by the site's present configuration, IP's are anonymized and Google Analytics AdSense features are disabled in compliance with GDPR - scripts still fire but functionality is limited - unless/until the user sets a cookie to indicate acceptance).

**Long answer:** per [FreeUKGen/FreeUKGenealogy/issues/26](https://github.com/FreeUKGen/FreeUKGenealogy/issues/26) I took an inventory of existing Google Analytics configuration in preparation for use of Google Tag Manager - it appears as though the `loadGoogleScripts` and `gaOptout` functions are not referenced in the codebase (likewise there is no element with the ID `acceptAnalytic` - resulting in a Javascript error for all pages), however, `loadAnalyticScripts` does check `acceptAnalyticIsSet` and will anonymize IP if user has not set the acceptance cookie for Analytics and `loadAdsenseScripts` will similarly disable `allowAdFeatures` for Google Analytics if user has not set the acceptance cookie for AdSense.

**Comment by richpomfret** [comment]
Done a little while ago. I wonder whether this will be on hold until the Wordpress update has been made.

**Comment by arswright-alt-uk** [comment]
@richpomfret 

> I wonder whether this will be on hold until the Wordpress update has been made.

I'd like to address the [Craft Transfer to AWS (FreeUKGenealogy/issues/5)](https://github.com/FreeUKGen/FreeUKGenealogy/issues/5) and any relatively-minor Craft fixes as prioritized by @DeniseColbert immediately with the goal of establishing as much of the desired functionality and content as possible before a feature freeze to support [WordPress Migration (FreeUKGenealogy/issues/44)](https://github.com/FreeUKGen/FreeUKGenealogy/issues/44) - will proceed through backlog as prioritized.

**Comment by richpomfret** [comment]
Thanks for clarifying and sounds good to me @arswright-alt-uk

**Comment by richpomfret** [comment]
I think this now needs to go 'on ice' until we make a decision on Wordpress rollout (considering the resourcing situation for this). It may well be desirable for us to go ahead and add in TAG manager when we roll it out on our other projects too (make it one big project, once we get to that stage in BMD2).

**Comment by richpomfret** [comment]
We now have a new volunteer to help with this. However, we need to discuss how we might rollout Wordpress to replace Craft - which ideally should happen before GTA is implemented. IF too problematic, I would suggest moving ahead with using custom templates for GTA in the meantime (in order to handle GDPR settings with cookies).
